### PR TITLE
TEST: Vuejs import styleguide - test with x-form login

### PR DESCRIPTION
**Pull request by fuentesloic**
capture d ecran 2017-09-18 a 12 18 46-fullpage

### CODE REVIEW

File: templates/auth/login.html [hunk 1]
```diff
@@ -8,78 +8,45 @@
...
-    margin-bottom: 0.5rem;
-  }
-
-  input {
-    margin-bottom: 2rem;
-  }
-  form button {
-    margin: 0;
-  }
+form > div > label[data-v-6e1bd991] {
```
Comment (vinyll) [inline templates/auth/login.html:26]: Nope, don't do that. The `data-v-6e1bd991` may change.

File: templates/auth/login.html [hunk 2]
```diff
@@ -8,78 +8,45 @@
...
-  <div>
-    <label for=password>Password</label>
-    <input type=password id=password name=password required>
-  </div>
-  <button type=submit>Log in</button>
-  <p class=alternative><a href={{ url_for('signup') }}>I don't have an account yet!</a></p>
-</form>
+<section id=body>
+  {% block login %}
+  <x-form class=login action={{ url_for('login') }} method=post>
```
Comment (vinyll) [inline templates/auth/login.html:33]: for actual usage outside the demo you may want to say `<form is=x-form …>`.

File: templates/auth/login.html [hunk 3]
```diff
@@ -8,78 +8,45 @@
...
-  }
-
-  input {
-    margin-bottom: 2rem;
-  }
-  form button {
-    margin: 0;
-  }
+form > div > label[data-v-6e1bd991] {
+  width: 9rem
```
Comment (vinyll) [inline templates/auth/login.html:27]: Why do you need this BTW?

File: templates/auth/login.html [hunk 4]
```diff
@@ -8,78 +8,45 @@
...
 }
 </style>
 
-{% block login %}
-<form action={{ url_for('login') }} method=post>
-  {% if errors %}
-    <ul>
-    {% for error in errors %}
```
Comment (vinyll) [inline templates/auth/login.html:67]: You still need this, don't you?

File: templates/auth/login.html [hunk 5]
```diff
@@ -8,78 +8,45 @@
...
-  <p class=alternative><a href={{ url_for('signup') }}>I don't have an account yet!</a></p>
-</form>
+<section id=body>
+  {% block login %}
+  <x-form class=login action={{ url_for('login') }} method=post>
+    <div class=form-row>
+      <label for=email>Email</label>
+      <input type=email name=email id=email>
+    </div>
+    <div class=form-row>
```
Comment (vinyll) [inline templates/auth/login.html:38]: `class=form-row` is actually not required: https://github.com/neomad-team/styleguide/blob/vuejs-neomad/src/components/x-form.vue#L19

We should adapt the styleguide demo also.

File: templates/auth/login.html [hunk 6]
```diff
@@ -8,78 +8,45 @@
...
-  </div>
-  <button type=submit>Log in</button>
-  <p class=alternative><a href={{ url_for('signup') }}>I don't have an account yet!</a></p>
-</form>
+<section id=body>
+  {% block login %}
+  <x-form class=login action={{ url_for('login') }} method=post>
+    <div class=form-row>
+      <label for=email>Email</label>
+      <input type=email name=email id=email>
```
Comment (vinyll) [inline templates/auth/login.html:36]: `required`

File: templates/auth/login.html [hunk 7]
```diff
@@ -8,78 +8,45 @@
...
+    </div>
+    <div class=form-row>
+      <label for=password>Password</label>
+      <input type=password id=password name=password required>
+    </div>
+    <input is=x-button class=primary type=submit name=login value=Login>
```
Comment (vinyll) [inline templates/auth/login.html:42]: Why does the text look so small on your screenshot

File: templates/auth/login.html [hunk 8]
```diff
@@ -8,78 +8,45 @@
...
+    <hr>
+    <input type=submit name=signup value="Create an account" class=default>
+  </x-form>
+  {% endblock %}
+</section>
 {% endblock %}
+
+{% block extra_js %}
+<script src={{ url_for('static', filename='dist/build.js') }}></script>
```
Comment (vinyll) [inline templates/auth/login.html:51]: This shall probably end up in the layout.
We may also want to rename it to _dist/components.js_ or something (not sure neither about a _dist_ folder for Neomad.org of _js_ folder, or elsewhere).

Comment (vinyll) [reply templates/auth/login.html:51]: We shall probably also need a _styleguide.css_ or something.

Comment (vinyll) [reply templates/auth/login.html:51]: be aware that components call an external css! (how should we make this dependency??): https://github.com/neomad-team/styleguide/blob/vuejs-neomad/src/components/x-form.vue#L2
Not good 🤔

**Review by vinyll — CHANGES_REQUESTED** [review]

**Comment by vinyll** [comment]
I ❤️ed this result but I don't think we'll use it to go for css `@import`.
Should we close this PR @fuentesloic / @newick?

**Comment by newick** [comment]
@vinyll if `@import`is the easiest way, i'm in

**Comment by vinyll** [comment]
Closed for using native CSS instead.

**Pull request closed without merging**